src/dirbrowserformaction.cpp

```diff
@@ -0,0 +1,372 @@
+#include "dirbrowserformaction.h"
+
+#include <algorithm>
+#include <cstring>
+#include <curses.h>
+#include <dirent.h>
+#include <grp.h>
+#include <iomanip>
+#include <iostream>
+#include <pwd.h>
+#include <sstream>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "fmtstrformatter.h"
+#include "logger.h"
+#include "strprintf.h"
+#include "utils.h"
+#include "view.h"
+
+namespace newsboat {
+
+DirBrowserFormAction::DirBrowserFormAction(View* vv,
+	std::string formstr,
+	ConfigContainer* cfg)
+	: FormAction(vv, formstr, cfg)
+{
+	// In DirBrowser, keyboard focus is at the input field, so user can't
+	// possibly use 'q' key to exit the dialog
+	KeyMap* keys = vv->get_keys();
+	keys->set_key(OP_QUIT, "ESC", id());
+	vv->set_keymap(keys);
+}
+
+DirBrowserFormAction::~DirBrowserFormAction() {}
+
+char get_filetype(mode_t mode)
+{
+	static struct FlagChar {
+		mode_t flag;
+		char ftype;
+	} flags[] = {{S_IFREG, '-'},
+		{S_IFDIR, 'd'},
+		{S_IFBLK, 'b'},
+		{S_IFCHR, 'c'},
+		{S_IFIFO, 'p'},
+		{S_IFLNK, 'l'},
+		{S_IFSOCK, 's'},
+		{0, 0}};
+
+	for (unsigned int i = 0; flags[i].flag != 0; i++) {
+		if ((mode & S_IFMT) == flags[i].flag)
```

Comment:
I noted in one of the earlier reviews that we should always apply the boyscout rule, even when copy-pasting. Please do so here, and add the curly braces around the body of this `if`.

There are more of those further down. Please look through this PR and fix them all.